cpp/src/arrow/compute/row/grouper.h

```diff
@@ -30,6 +30,49 @@
 namespace arrow {
 namespace compute {
 
+/// \brief A segment of contiguous rows for grouping
+struct ARROW_EXPORT GroupingSegment {
```

Comment:
Can you explain what the four variables mean here

Reply:
Done.